src/parser/mod.rs

```diff
@@ -15017,6 +15075,13 @@ impl<'a> Parser<'a> {
         }
     }
 
+    fn parse_return(&mut self) -> Result<Statement, ParserError> {
+        let expr = self.parse_expr()?;
```

Comment:
Does this need to be a maybe_parse, like how parse_raise_stmt does it for Exprs? The approach I picked for the discussion of RETURN variants above was to only do what's needed for _this_ PR & _these_ tests, which always have an Expr after the return.

So I think perhaps what we'd need in a followup PR is:
1. test case for bare RETURN
2. update this to be use "maybe" logic to find an expr (perhaps we'd like `maybe_expr` to wrap up that pattern?)

Reply:
Maybe we can remove the return statement bits from this PR and tackle it separately as its own PR?

Otherwise indeed the expectation would be that this uses maybe_parse and we have the test cases in place as part of this PR

Reply:
> Maybe we can remove the return statement bits from this PR and tackle it separately as its own PR?

I don't think this is possible, since SQL Server requires that functions have a return, so create function goes together with return. I'll take another pass on the PR though

Reply:
Ah alright, in that case I think its only the test cases missing for the return statement we can add those to common.rs I imagine

Reply:
I've added a test case for a bare `RETURN` to sqlparser_common.rs, although I'm unsure how useful it is. At least for SQL Server, a top level return _must_ be bare (eg, can't return anything). Therefore, the more useful tests are where return is inside a function body, trigger body, etc. And we have those tests in their respective branches.